Mathlib/Computability/DFA.lean

```diff
@@ -166,4 +173,39 @@ theorem pumping_lemma [Fintype σ] {x : List α} (hx : x ∈ M.accepts)
   rwa [mem_accepts, evalFrom_of_append, evalFrom_of_append, h, hc]
 #align DFA.pumping_lemma DFA.pumping_lemma
 
+section equivOfStates
+
+variable {σ' : Type*} (f : σ ≃ σ') (M : DFA α σ)
+
+/-- Lift an equivalence on states to an equivalence on DFAs. -/
+def equivOfStates : DFA α σ ≃ DFA α σ' where
```

Comment:
```suggestion
def congr : DFA α σ ≃ DFA α' σ' where
```
or something like that. Can you also define `map : DFA α σ → DFA α' σ'`?

Reply:
That's interesting because `σ` is both an input and output, but `α` is an input only. So I guess we only need `α' ↪︎ α`?